sockeye/image_captioning/inference.py

```diff
@@ -141,17 +141,20 @@ def translate(self, trans_inputs: List[TranslatorInput]) -> List[TranslatorOutpu
 
         return results
 
-    def _get_inference_input(self, image_paths: List[str]) -> Tuple[mx.nd.NDArray, int, List[Optional[constrained.RawConstraintList]]]:
+    def _get_inference_input(self, image_paths: List[str]) -> Tuple[mx.nd.NDArray, int, List[Optional[constrained.RawConstraintList]], mx.nd.NDArray]:
         """
-        Returns NDArray of images and corresponding bucket_key.
+        Returns NDArray of images and corresponding bucket_key and an NDArray of maximum output lengths for each sentence in the batch.
 
         :param image_paths: lists of image paths.
-        :return NDArray of images paths, bucket key, a list of raw constraint lists.
+        :return NDArray of images paths, bucket key, a list of raw constraint lists, an NDArray of maximum output lengths.
         """
         ## TODO(bazzanil): support constraints
         raw_constraints = [None for x in range(self.batch_size)]  # type: List[Optional[constrained.RawConstraintList]]
+        max_output_lengths = [] # type: List[int]
+        for i in range(len(image_paths)):
+            max_output_lengths.append(self.models[0].get_max_output_length(len(image_paths[i])))
         images = self.data_loader(image_paths, self.source_image_size)
-        return mx.nd.array(images), 0, raw_constraints
+        return mx.nd.array(images), 0, raw_constraints, mx.nd.array(max_output_lengths, ctx=self.context, dtype='int32')
```

Comment:
actually here previously we passed the `0` as source length to `get_max_output_length`. Therefore you should do something like:
```
max_input_len = 0
max_output_lengths = mx.nd.array([self.models[0].get_max_output_length(max_input_len)] * len(image_paths[i]), ...)
return mx.nd.array(images), max_input_len, raw_constraints, max_output_lens
```

The actual value is not actually important as it is anyway overwritten through the `forced_max_output_len` flag when captioning, but just to stay consistent with the rest of of the code